module/zfs/metaslab.c

```diff
@@ -325,19 +354,22 @@ static void metaslab_flush_update(metaslab_t *, dmu_tx_t *);
 static unsigned int metaslab_idx_func(multilist_t *, void *);
 static void metaslab_evict(metaslab_t *, uint64_t);
 static void metaslab_rt_add(range_tree_t *rt, range_seg_t *rs, void *arg);
-#ifdef _METASLAB_TRACING
```

Comment:
Enabling these kstats and not the full tracing by default sounds great.  But then let's enable all of the kstats wrapped by this undefined macro below.  It looks like we could remove the macro entirely if we changed the default to `metaslab_trace_enabled = B_FALSE`?  Or does this still result in significant overhead?

Reply:
> But then let's enable all of the kstats wrapped by this undefined macro below.

I don't quite follow, could you be more specific?  Are you talking about `metaslabstat_df_find_under_floor`?

> It looks like we could remove the macro entirely if we changed the default to metaslab_trace_enabled = B_FALSE? Or does this still result in significant overhead?

I think that would be fine.  The only possible concern is that we may be frequently bumping `metaslabstat_df_find_under_floor`, which I left under `#ifdef _METASLAB_TRACING`.  While I don't think this would impact performance, it might be hard to verify that.  I'd be OK with simply removing it.  Longer term, it would be nice to have some infrastructure for easily creating scalable (per-CPU) kstats (like a lighter weight aggsum), so that we wouldn't have to worry about the cost of incrementing them.  Any thoughts @pcd1193182 ?

Reply:
Yes, I was referring to `metaslabstat_df_find_under_floor` which I noticed was still wrapped by `_METASLAB_TRACING`.  I think removing it would be preferable if we have any real concerns about it impacting performance.  Though it doesn't look like it would be too bad.

We can easily enough remove the remaining `_METASLAB_TRACING` usage around the tracing in another PR if you like.  This just seemed like a good opportunity to get rid of that lingering bit of ugliness.

Reply:
My opinion is that we can probably remove find_under_floor and enable all the other stuff under `METASLAB_TRACING` by default. If we had a lighter-weight aggsum that would make enabling find_under_floor better, but in that case we should also probably switch most of our kstats over to that.

Reply:
Happy to do the cleanup in this PR.